### PR TITLE
[proof-new] Updating SAT proof to use MACRO_RESOLUTION

### PR DESCRIPTION
**Pull request by HanielB**

### CODE REVIEW

**Comment by HanielB** [comment]
Note that this PR should only go in after #5612. It also is a super set of that one, so one only needs to review the changes to `prop/sat_proof_manager.*`

**Review by nafur — COMMENTED** [review]
A few suggestions.

File: src/prop/sat_proof_manager.cpp [hunk 1]
```diff
@@ -82,19 +82,26 @@ void SatProofManager::startResChain(const Minisat::Clause& start)
     Trace("sat-proof") << "\n";
   }
   d_resLinks.push_back(
-      std::pair<Node, Node>(getClauseNode(start), Node::null()));
+      std::make_tuple(getClauseNode(start), Node::null(), true));
 }
```
[the remark this answers is not in the document]
Comment (HanielB) [reply src/prop/sat_proof_manager.cpp:86]: Neat!

**Comment by ajreynol** [comment]
This should come after the other one right?

**Comment by HanielB** [comment]
Right. I left the `do not merge` tag for that.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**